odoo/orm/environments.py

```diff
@@ -162,11 +162,7 @@ def ref(self, xml_id: str, raise_if_not_found: bool = True) -> BaseModel | None:
         )
 
         if res_model and res_id:
-            record = self[res_model].browse(res_id)
-            if record.exists():
-                return record
-            if raise_if_not_found:
-                raise ValueError('No record found for unique ID %s. It may have been deleted.' % (xml_id))
+            return self[res_model].browse(res_id)
```

Comment:
Why do you change this in this PR?
AFAIK, the goal is to raise early here if the referenced record does not exist. The `_xmlid_to_res_model_res_id` raises only if the ir.model.data record is not found.

Reply:
I can be as it is, I just found it has not much to do with the description of the PR.